controllers/managedocs_controller.go

```diff
@@ -1067,6 +1085,40 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) getPrometheusRemoteWriteSpec() (*promv1.RemoteWriteSpec, error) {
+	if err := r.get(r.rhobsSecret); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, fmt.Errorf("rhobs secret missing: %v", err)
+		} else {
+			return nil, fmt.Errorf("unable to fetch rhobs secret: %v", err)
+		}
+	}
+	rhobsSecretData := r.rhobsSecret.Data
+	if _, found := rhobsSecretData[rhobsConfigIdName]; !found {
+		return nil, fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsConfigIdName)
+	}
+
+	if _, found := rhobsSecretData[rhobsConfigSecretName]; !found {
+		return nil, fmt.Errorf("rhobs secret does not contain a value for key %v", rhobsConfigSecretName)
+	}
+
+	rhobsAudience, found := rhobsSecretData["rhobs-audience"]
+	if !found {
+		return nil, fmt.Errorf("rhobs secret does not contain a value for key rhobs-audience")
+	}
+
+	remoteWriteSpec := templates.PrometheusRemoteWriteSpecTemplate.DeepCopy()
```

Comment:
This should come from the prom template that was already deep copied, we do not do templates for subsection of resources